### PR TITLE
Fix Typescript errors

### PR DESCRIPTION
**Pull request by RatoX**
Fix wrong generic use and improve tests to TypeDefinition.

### CODE REVIEW

File: index.test.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import {TableMap, TableRect} from 'prosemirror-tables';
```
Comment (marduke182) [inline index.test.ts]: Should we use `../src` instead?

Comment (RatoX) [reply index.test.ts]: The types are not coming from the source code,  they are coming from the modules declaration `index.d.ts`

File: package.json [hunk 2]
```diff
@@ -46,6 +46,7 @@
     "typings-tester": "^0.3.2"
   },
   "scripts": {
+    "typecheck": "tsc --noEmit",
```
Comment (marduke182) [inline package.json]: should we run this on pre-build, no?

Comment (RatoX) [reply package.json]: We are running this on the tests

**Pull request merged**